homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,127 @@
+"""Config flow for myLeviton decora_wifi."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .common import (
+    DecoraWifiCommFailed,
+    DecoraWifiLoginFailed,
+    DecoraWifiPlatform,
+    decorawifisessions,
+)
+from .const import CONF_TITLE, DOMAIN
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class DecoraWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Decora Wifi config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.session = None
+        self.data = {}
+        self.entry = None
+
+    async def async_step_user(self, user_input=None, errors=None):
+        """Prompt for user input to setup decora_wifi."""
+
+        data_schema = {
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+        }
+
+        if user_input is None:
+            # Show the form requesting the Username and Password for Decora Wifi
+            return self.async_show_form(
+                step_id="user", data_schema=vol.Schema(data_schema)
+            )
+
+        # Update flow state
+        self.data.update(user_input)
+
+        # Get user from existing entry and abort if already setup.
+        self.entry = await self.async_set_unique_id(self.data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        try:
+            # Attempt to log in with the credentials provided by the user.
+            self.session = await DecoraWifiPlatform.async_setup_decora_wifi(
+                self.hass,
+                email=self.data[CONF_USERNAME],
+                password=self.data[CONF_PASSWORD],
+            )
+        except DecoraWifiLoginFailed:
+            if errors is None:
+                errors = {"base": "invalid_auth"}
+            else:
+                errors["base"] = "invalid_auth"
```

Comment:
Make errors an empty `dict` above this so you don't have to check for `None`

Reply:
That's actually how I had it before, and pylint [threw a warning](https://github.com/home-assistant/core/pull/51092/checks?check_run_id=3389236596) about [empty dictionary being a dangerous default](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html).

Reply:
Done in next commit.